src/shares/handle_mining_message.rs

```
@@ -44,10 +43,15 @@ pub async fn handle_mining_message<C>(
                 share_block.header.miner_share.workinfoid, share_block.header.miner_share.hash
             );
             share_block = chain_handle.setup_share_for_chain(share_block).await;
+            let share_block_clone = share_block.clone();
             if let Err(e) = chain_handle.add_share(share_block).await {
                 error!("Failed to add share: {}", e);
                 return Err("Error adding share to chain".into());
             }
+            //Send INV message to the network
+            if let Err(e) = _swarm_tx.send(SwarmSend::Inv(share_block_clone)).await {
+                error!("Failed to send INV message to swarm: {}", e);
```

We should return an error from here as the sending failed, and then test for the error result in the test you added.